modules/internal/DefaultSparse.chpl

```diff
@@ -149,15 +154,15 @@ module DefaultSparse {
     }
 
     proc dsiLast {
-      return _indices[_nnz];
+      return _indices[_nnz-1];
     }
 
     proc add_help(ind) {
       // find position in nnzDom to insert new index
       const (found, insertPt) = find(ind);
 
       // if the index already existed, then return
-      if (found) then return 0;
+      if (found) then return 0;  // ?fromMMS: magic number for dsiAdd retval?
```

Comment:
I see this as more like saying "I have added 0 indices" to its caller. Check `bulkAdd_help` implementations, for example. They return number of indices added.